packages/core/core/src/Graph.js

```diff
@@ -72,39 +78,27 @@ export default class Graph<TNode: Node, TEdgeType: string | null = null> {
     return edges;
   }
 
-  addNode(node: TNode): TNode {
-    let existingNode = this.nodes.get(node.id);
-    if (existingNode) {
-      existingNode.value = node.value;
-    }
-    this.nodes.set(node.id, existingNode ?? node);
-    return node;
+  addNode(node: TNode): NodeId {
+    let id = toNodeId(this.nextNodeId++);
+    this.nodes.set(id, node);
+    return id;
   }
 
-  hasNode(id: string): boolean {
+  hasNode(id: NodeId): boolean {
     return this.nodes.has(id);
   }
 
-  getNode(id: string): ?TNode {
+  getNode(id: NodeId): ?TNode {
     return this.nodes.get(id);
   }
 
-  setRootNode(node: TNode): void {
-    this.addNode(node);
-    this.rootNodeId = node.id;
-  }
```

Comment:
Do we want to have a `graph.setRootNodeId(id: NodeId)` method instead of doing `graph.rootId = ...`?